lib/os/fdtable.c

```diff
@@ -522,6 +527,16 @@ int zvfs_ioctl(int fd, unsigned long request, va_list args)
 	return fdtable[fd].vtable->ioctl(fdtable[fd].obj, request, args);
 }
 
+int zvfs_feof(FILE *file)
+{
+	struct fd_entry *entry = (struct fd_entry *)file;
+
+	if (!IS_ARRAY_ELEMENT(fdtable, entry)) {
+		return -1;
+	}
+
+	return ((entry->flags & ZVFS_FLAG_EOF) != 0) ? 1 : 0;
```

Comment:
Okay, writing tests will be a bit tricky since the other functions aren't defined yet. I could set the indicator in `zvfs_rw` which I'm guessing is what functions like `fgetc` will wrap around.

Reply:
> There are some additional criteria to consider here:
> 
>     * What if the file is not open?
> 
>     * What if another operation is currently writing `flags`?
> 
>     * How can we verify that all of this works?
> 
> 
> I would suggest writing some tests as well.

Okay, then I could probably set the indicator in zvfs_rw and test it?

Reply:
@DanTGL

> * What if the file is not open?

Please use `_check_fd()`

> * What if another operation is currently writing `flags`?

Please lock `fd_entry.mutex` 

> * How can we verify that all of this works?
>
> I would suggest writing some tests as well.

Please write some tests